rustic-flycheck.el

```diff
@@ -68,7 +68,21 @@ Calls `cargo metadata --no-deps --manifest-path MANIFEST
 --format-version 1', parses and collects the targets for the
 current workspace, and returns them in a list, or nil if no
 targets could be found."
-  (let ((cargo (funcall flycheck-executable-find "cargo")))
+  (let ((process-output-as-json
+         (lambda (program &rest args)
+           (with-temp-buffer
+             (let ((code-or-signal (apply 'call-process program nil '(t nil) nil args)))
+               (unless (equal code-or-signal 0)
+                 ;; Prevent from displaying "JSON readtable error".
+                 (let* ((args (combine-and-quote-strings (cons program args)))
+                        (error-message (if (typep code-or-signal 'string)
```

Comment:
Out of curiosity, are there advantages using `typep` over `stringp` ?

Reply:
I had forgotten `certain-type-p` functions. `stringp` is appropriate.

Reply:
I will fix it.